src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ScenarioRunResult.cs

```diff
@@ -45,7 +46,8 @@ public sealed class ScenarioRunResult(
     DateTime creationTime,
     IList<ChatMessage> messages,
     ChatResponse modelResponse,
-    EvaluationResult evaluationResult)
+    EvaluationResult evaluationResult,
+    int formatVersion = Defaults.ReportingFormatVersion)
```

Comment:
We actually want the version to be part of the signature like this?

Reply:
It is a valid question if we could somehow omit the version from appearing in the public API. However, there doesn't seem to be an easy way to avoid this since the version needs to be included as part of the serialization of the results (so that we can know what the original serialized version was in the case where we need to do any special translation during deserialization to the 'current' version).

Our report storage is basically a collection of `ScenarioRunResult`s - it is currently the root object in the tree of result types, and we don't have a more root / wrapping concept - so this seemed like the best place to store this version for now.

Reply:
It couldn't be `int? formatVersion = null`?